modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -304,15 +329,11 @@ def opts_for_int_type(self, int_type):
         -------
         tuple
         """
-        if int_type is np.int8:
-            return "TINYINT", 3
-        if int_type is np.int16:
-            return "SMALLINT", 5
-        if int_type is np.int32:
-            return "INTEGER", 10
-        if int_type in (np.int64, int):
-            return "BIGINT", 19
-        raise NotImplementedError(f"Unsupported integer type {int_type.__name__}")
+        try:
+            assert _warn_if_unsigned(int_type)
```

Comment:
What is `assert` used here for?

Reply:
Just to do not run this code, if assertions are disabled. It could slightly improve performance and eliminate unnecessary warnings.

Reply:
wait, could you explain the motivation for all of these `assert`s again?

> It could slightly improve performance

is it really that significant?

> eliminate unnecessary warnings

we would better control the warnings by explicitly filtering them rather than noticing they suddenly appear/disappear when we switch off/on assertions

IMO these assertion looks more confusing than really helpful